### ISSUE TITLE
Specmatic should throw error instead of did not run when schema is invalid.

### ISSUE THREAD

**Issue by barath121** [specmatic/specmatic#1065]
**Description**
Specmatic should throw error instead of did not run when schema is invalid.

**Steps to reproduce**
1. Write a request body with schema of type array.
2. Add invalid examples of type object.
3. Run test.

**Expected behaviour **
Should throw an error that schema/examples invalid.

**Comment by yogeshnikam671** [comment]
Hi @barath121, thanks for reporting this issue. We will have a look at it and get back.

**Comment by yogeshnikam671** [comment]
Hi @barath121, I am assuming you are not adding the corresponding named example in the response. Is that right?
For example, you have a specification that looks like this, where the named example `TEST` is missing from responses -
```
  /products/test:
    post:
      summary: POST /test
      requestBody:
        required: true
        content:
          application/json:
            schema:
              type: array
              items:
                type: string
            examples:
              TEST:
                value:
                  first: firstValue
                  second: secondValue
      responses:
        "200":
          description: POST /test
          content:
            text/plain;charset=UTF-8:
              schema:
                type: string 
```

If yes, then this is an existing issue. 
In such cases, Specmatic does not make use of the named example present in the request and generates a random request on its own. 
It is already getting tracked as part of [#1063](https://github.com/znsio/specmatic/issues/1063)


**If we add the corresponding named example `TEST` in the responses, the test will fail with the error** - 
```
Error:
        Contract expected json array but found value {"first":"firstValue","second":"secondValue"} 
```

Although, the point you are raising is valid. The schema validation has to happen first before the generation of the tests.
We will discuss this within the team and get back on the conclusion.

**Comment by yogeshnikam671** [comment]
@barath121  Can you confirm which version of Specmatic you are using ? Are you on latest (1.3.11) ?

**Comment by yogeshnikam671** [comment]
This is resolved in release 1.3.12. 
Closing for now. Feel free to reopen if necessary.
